oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/LucenePropertyIndex.java

```diff
@@ -1551,7 +1551,7 @@ private static Query tokenToQuery(String text, String fieldName, PlanResult pr,
         if (FieldNames.FULLTEXT.equals(fieldName)) {
             Query subQuery = new BooleanQuery();
             if (pr.indexDefinition.isDynamicBoostLiteEnabled()) {
-                subQuery = new TermQuery(new Term(FieldNames.SIMILARITY_TAGS, text));
+                subQuery = new TermQuery(new Term(FieldNames.SIMILARITY_TAGS, text.toLowerCase()));
```

Comment:
@fabriziofortino here is place I'd like to change for fixing case insensitive query over dynamicBoostLite

Reply:
@FrancoisZhang that's a possibility but we are going to apply other possible transformations that the analyzer does.

Could you instead try to modify this code with:

```java
subQuery = tokenToQuery(text, FieldNames.SIMILARITY_TAGS, analyzer);
```

In this way, we use the lucene analyzer that applies a number of transformations. I think this was the main issue with dynamic boost lite (my bad). Could you try this and verify the other tests work? Thanks

Reply:
@fabriziofortino quick test seems doesn't work. can you check latest code? I use a new created method to make it work the same as dynamicBoost query. which is actually implemented in AEM DAM code base. new method I create here is try to mimic with that and make them work as the same way.

Reply:
@fabriziofortino  I verified it again, tokenToQuery does great to make it fully supporting the lucene fulltext query syntax. I made another commit to use that instead and added some tests to verify it's query ability. It should make more sense here. Only concern is it will make it doesn't match the behavior of DynamicBoost in Lucene and Elastic. More like a product decision here though.